### PR TITLE
bpo-40246: Revert reporting of invalid string prefixes

### PR DESCRIPTION
**Pull request by lysnikolaou**
Due to backwards compatibility concerns, that keywords immediately
followed by a string without whitespace between them
(like in `bg="#d00" if clear else"#fca"`) will fail to parse,
GH-19476 has to be reverted.

https://bugs.python.org/issue40246

### CODE REVIEW

**Review by gvanrossum — APPROVED** [review]

**Review by gvanrossum — CHANGES_REQUESTED** [review]
Please also delete the NEWS item of the reverted change (Misc/NEWS.d/next/Core and Builtins/2020-04-11-17-52-03.bpo-40246.vXPze5.rst).

**Comment by lysnikolaou** [comment]
Can't find the news file. There is not file with `bpo-40246` on master according to Github and to my local clone.

**Comment by hroncok** [comment]
That news file has already been merged to the changelog when 3.9.0a6 was released. I think this should explicitly add a new entry about the revert.

File: Misc/NEWS.d/next/Core and Builtins/2020-05-03-23-28-11.bpo-40246.c1D7x8.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Reporting a specialised error message for invalid string prefixes, which was introduced in #19888, is being reverted due to backwards compatibility concerns for strings that immediately follow a reserved keyword without whitespace between them. Constructs like `bg="#d00" if clear else"#fca"` were failing to parse, which is not an acceptable breakage on such short notice.
```
Comment (hroncok) [inline Misc/NEWS.d/next/Core and Builtins/2020-05-03-23-28-11.bpo-40246.c1D7x8.rst]: 19888 is this one

Comment (lysnikolaou) [reply Misc/NEWS.d/next/Core and Builtins/2020-05-03-23-28-11.bpo-40246.c1D7x8.rst]: Whoops. Thanks for the catch!

**Comment by lysnikolaou** [comment]
I have made the requested changes; please review again

File: Misc/NEWS.d/next/Core and Builtins/2020-05-03-23-28-11.bpo-40246.c1D7x8.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Reporting a specialised error message for invalid string prefixes, which was introduced in GH-19476, is being reverted due to backwards compatibility concerns for strings that immediately follow a reserved keyword without whitespace between them. Constructs like `bg="#d00" if clear else"#fca"` were failing to parse, which is not an acceptable breakage on such short notice.
```
Comment (pablogsal) [inline Misc/NEWS.d/next/Core and Builtins/2020-05-03-23-28-11.bpo-40246.c1D7x8.rst]: Please, mention the bpo number instead of the PR number (GH-19476)

Comment (pablogsal) [inline Misc/NEWS.d/next/Core and Builtins/2020-05-03-23-28-11.bpo-40246.c1D7x8.rst]:
```suggestion
Reporting a specialised error message for invalid string prefixes, which was introduced in GH-19476, is being reverted due to backwards compatibility concerns for strings that immediately follow a reserved keyword without whitespace between them, identifying constructs like `bg="#d00" if clear else"#fca"` incorrectly invalid.
```

Comment (gvanrossum) [reply Misc/NEWS.d/next/Core and Builtins/2020-05-03-23-28-11.bpo-40246.c1D7x8.rst]: Honestly I like @lysnikolaou's version better.

**Review by gvanrossum — APPROVED** [review]

File: Misc/NEWS.d/next/Core and Builtins/2020-05-03-23-28-11.bpo-40246.c1D7x8.rst (hunk 2 above)
Comment (pablogsal) [reply Misc/NEWS.d/next/Core and Builtins/2020-05-03-23-28-11.bpo-40246.c1D7x8.rst]: Fair enough, let's skip my suggestion then.

**Comment by hroncok** [comment]
I've backported this PR t Fedora's build of 3.9.0a6 and I still see:

```
  File "/usr/lib/python3.9/site-packages/dnf/comps.py", line 111
    return'.'.join(lcl)
         ^
SyntaxError: invalid string prefix
```

I will double check the patch was actually applied.

**Comment by hroncok** [comment]
It was a cache issue, it used an older, unpatched build. Sorry for the noise.

In fact, this PR makes the problem go away, as ecpected.

**Comment by lysnikolaou** [comment]
No `SyntaxError` when building this branch locally.

```
╰─ ./python.exe
Python 3.9.0a6+ (heads/master-dirty:64224a4727, May  3 2020, 23:23:39)
[Clang 11.0.0 (clang-1100.0.33.8)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> def f():
...     return'.'.join(lcl)
...
>>> f()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 2, in f
NameError: name 'lcl' is not defined
```

File: Misc/NEWS.d/next/Core and Builtins/2020-05-03-23-28-11.bpo-40246.c1D7x8.rst (hunk 2 above)
Comment (pablogsal) [inline Misc/NEWS.d/next/Core and Builtins/2020-05-03-23-28-11.bpo-40246.c1D7x8.rst]:
```suggestion
Reporting a specialised error message for invalid string prefixes, which was introduced in :issue:`40246`, is being reverted due to backwards compatibility concerns for strings that immediately follow a reserved keyword without whitespace between them. Constructs like `bg="#d00" if clear else"#fca"` were failing to parse, which is not an acceptable breakage on such short notice.
```

**Review by pablogsal — APPROVED** [review]

**Pull request merged**